src/clr-angular/data/datagrid/providers/state.provider.ts

```diff
@@ -102,17 +103,17 @@ export class StateProvider<T> {
     if (state.filters) {
       const activeFilters = this.filters.getActiveFilters();
       for (const filter of state.filters) {
-        let filterObject: ClrDatagridFilterInterface<T>;
+        let filterObject: SerializableFilter<T>;
         if (filter.hasOwnProperty('property') && filter.hasOwnProperty('value')) {
```

Comment:
This special case should be removed in favor of the new state interface of filters. The special case don't work properly because it creates new filters objects instead of updating the existing ones.

Reply:
New filter creation happens after this "if else" block, in this block we are converting the old filter prop value into the new filterstate object, which is essential, if we omit this block, the functionality will be broken. Can you specify the details of this refactor, because I can't see what exactly do you mean?

Reply:
Is missed the piece in the code where `const existing = activeFilters.findIndex(value => value.compatibleToState(filterObject.filterState)); ` is called even for the builtin filters.

So the code looks correct, but I would eliminate the special case anyway by modifying the state getter of the state provider to just emit the `filterState` for builtin string filters as well.

Reply:
This would break existing filters. But you need to check filters if the filters are `SerializableFilter` and if so, use the state getter. So instead of having this [special case](https://github.com/Qabel/clarity/pull/1/files/83f3e3795d3bb71aa0c2bee995299be82659f290..3fc3f82b78a7d2367019d8298631576c4bd6a28f#diff-fc88a77fd471217bec59eef4e132ae52R64), you can use the state getter for all serializable filters.

Reply:
Yes, or by checking the interface manually. The abstract class would be a nicer solution in that regard, but I don't know which problems this creates in the long term. I guess it would be a good start and then you can see what the clarity maintaners think about that approach.

Reply:
Just checked since getActiveFilters() already returns `SerializableFilter`s, we don't need to check the type, special case is eliminated. Check the last commit.